agent/runner/jobs/mongodb_restore_job.go

```diff
@@ -172,44 +177,25 @@ func (j *MongoDBRestoreJob) Run(ctx context.Context, send Send) error {
 	return nil
 }
 
-func (j *MongoDBRestoreJob) findSnapshot(ctx context.Context, snapshotName string) (*pbmSnapshot, error) {
+func (j *MongoDBRestoreJob) findCurrentSnapshot(ctx context.Context, snapshotName string) (*pbmSnapshot, error) {
 	j.l.Info("Finding backup entity name.")
 
-	var list pbmList
-	ticker := time.NewTicker(listCheckInterval)
-	defer ticker.Stop()
-
-	checks := 0
-	for {
-		select {
-		case <-ticker.C:
-			checks++
-			if err := execPBMCommand(ctx, j.dbURL, &list, "list"); err != nil {
-				return nil, err
-			}
-
-			if len(list.Snapshots) == 0 {
-				j.l.Debugf("Try number %d of getting list of artifacts from PBM is failed.", checks)
-				if checks > maxListChecks {
-					return nil, errors.Wrap(ErrNotFound, "got no one snapshot")
-				}
-				continue
-			}
+	snapshots, err := getSnapshots(ctx, j.l, j.dbURL)
+	if err != nil {
+		return nil, err
+	}
 
-			// Old artifacts don't contain pbm backup name.
-			if snapshotName == "" {
-				return &list.Snapshots[len(list.Snapshots)-1], nil
-			}
+	// Old artifacts don't contain pbm backup name.
+	if snapshotName == "" {
+		return &snapshots[0], nil
```

Comment:
We changed command to get snapshots, and that new one returns snapshots in opposite order.